src/App/Repository/PublicDiscussionRepository.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Repository;
+
+use Doctrine\ORM\EntityRepository;
+
+class PublicDiscussionRepository extends EntityRepository
+{
+    public function findGlobalDiscussions()
+    {
+        $discussions = $this->findAll();
+        $globalDiscussions = [];
+
+        foreach ($discussions as $discussion) {
+            if ($discussion->globalDiscussion()) {
+                $globalDiscussions[] = $discussion;
```

Comment:
Mmm. You should only retrieve global discussions, and not add useless requests by retrieving all discussions.